web/modules/custom/product_uploader/product_uploader.module

```diff
@@ -0,0 +1,27 @@
+<?php
+
+/**
+ * @file
+ * Contains hook for Product Uploader module.
+ */
+
+declare(strict_types=1);
+
+use Drupal\Core\Entity\EntityInterface;
+
+/**
+ * Implements hook_entity_presave().
+ */
+function product_uploader_entity_presave(EntityInterface $entity) {
+  if ($entity->bundle() === 'user') {
+    $mail = $entity->get('mail')->value;
+    if (strpos($mail, '+productuploader')) {
+      $entity->addRole('product_uploader');
```

Comment:
In the else block you check if this user has the `product_uploader` role, but you don't check it here. Is not necessary to check for the role, or did you miss something?

Reply:
It's not part of the task so I removed it.